### PR TITLE
Enable admins to promote a facility match to become the canonical facility

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

This PR enables admins to promote a facility match to become the canonical facility. I updated the split match user interface to become `Adjust` match, and added a `Promote` button for each of the matches.

Clicking `Promote` will call the new API endpoint which updates the facility to use the list item for the selected match as its `created_from` & also updates its name, address, and location data for the new list item. It also writes an entry to the processing results log. Finally, it sends back the facility data to the adjust match screen, updating everything so that it is reflective of the new facility data.

Connects #639 

## Demo

Screen Shot 2019-07-18 at 5 45 08 PM

Screen Shot 2019-07-18 at 5 45 13 PM

## Testing Instructions

- using the standard set of facility data, sign in as an administrator
- merge some facilities together and copy the OAR ID
- paste that OAR ID into the new adjust path's component, then verify that you can
   - promote a match to become the canonical facility, seeing the changes reflected in the UI
   - continue to split facility matches using the existing apparatus.

## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [ ] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by kellyi** [comment]
I still need to make an addition to the CHANGELOG for this.

**Comment by jwalgran** [comment]
Looking at this now.

**Comment by kellyi** [comment]
I remembered the q I was trying to recall earlier:

Should I also adjust the match statuses and/or confidence scores in some way for the two exchanged matches?

File: src/django/api/views.py [hunk 1]
```diff
@@ -1057,14 +1057,86 @@ def split(self, request, pk=None):
...
+
+        try:
+            facility = Facility \
+                .objects \
+                .get(pk=pk)
+
+            match = FacilityMatch \
+                .objects \
+                .get(pk=match_id)
+
```
Comment (jwalgran) [inline src/django/api/views.py]: Consider raising `BadRequestException` in these cases
- `match.facility != facility`
We should only allow promotion if the match already points to the facility
- `facility.created_from == match.facility_list_item`
We don't need to promote if the match is already the "winner"

File: src/django/api/views.py [hunk 2]
```diff
@@ -1057,14 +1057,86 @@ def split(self, request, pk=None):
...
+            previous_created_from_id = facility.created_from.id
 
-        return Response('ok')
+            facility.name = match.facility_list_item.name
+            facility.address = match.facility_list_item.address
+            facility.country_code = match.facility_list_item.country_code
+            facility.location = match.facility_list_item.geocoded_point
+            facility.created_from = match.facility_list_item
+            facility.save()
```
Comment (jwalgran) [inline src/django/api/views.py]: Consider setting `facility.changeReason` before saving to something like "Promoted list item {new list item id} from {new list as string} over list item {previous list item id} from {previous list as string}"

File: src/django/api/views.py (hunk 1 above)
Comment (kellyi) [reply src/django/api/views.py]: Good idea -- I added these checks and also tests for them in 4c8f50d7ab258527c7915cdd825cbfccae7ccf5f

File: src/django/api/views.py (hunk 2 above)
Comment (kellyi) [reply src/django/api/views.py]: I added this in 5b5ce2bee073396f3fc9875aabadc7c59420dd36

**Pull request merged**